airflow/providers/google/cloud/hooks/dataflow.py

```diff
@@ -640,9 +659,8 @@ def start_python_dataflow(  # pylint: disable=too-many-arguments
         variables['job_name'] = name
         variables['region'] = location
 
-        def label_formatter(labels_dict):
-            return ['--labels={}={}'.format(key, value)
-                    for key, value in labels_dict.items()]
+        if 'labels' in variables:
+            variables['labels'] = ['{}={}'.format(key, value) for key, value in variables['labels'].items()]
```

Comment:
Why is this formatter different than the one from L521?

Could we move all label fomating to the place where dataflow job is triggered?

Reply:
It depends on the SDK that is used.  These two SDK require different argument formats.

We have three related methods.
* start_python_dataflow  
* start_java_dataflow
* _start_dataflow
The first two methods are public and dependent on the SDK. They are responsible for introducing changes dependent on the SDK, e.g. environment preparation. 
_start_dataflow is an internal method. This starts the system process and supervises execution. I have the impression that this separation is helpful.

Reply:
OK. Thank you for the explanation.